src/base/init/init.c

```diff
@@ -411,26 +411,26 @@ void low_mem_init(void)
     assert(dptr);
     /* we have an uncommitted hole up to phys_low */
     ptr += phys_low;
-    /* create non-identity mapping up to phys_low */
-    ptr2 = smalloc_topdown(&main_pool, phys_low);
-    assert(ptr2);
+    ptr2 = MEM_BASE32(memsize);
     if (config.cpu_vm_dpmi == CPUVM_KVM) {
       /* map dpmi+uncommitted space to kvm */
       int prot = PROT_READ | PROT_WRITE | PROT_EXEC;
       mmap_kvm(MAPPING_INIT_LOWRAM, phys_low, ptr2 - ptr, ptr, phys_low, prot);
     }
-    /* unused hole for alignment */
-    ptr2 += LOWMEM_SIZE + HMASIZE;
-    register_hardware_ram_virtual('U', DOSADDR_REL(ptr2), phys_rsv,
-	    LOWMEM_SIZE + HMASIZE);
-    /* register ext_mem and XMS alias memory for dpmi */
-    register_hardware_ram_virtual('X', LOWMEM_SIZE + HMASIZE, phys_rsv,
-	    DOSADDR_REL(ptr2));
+    /* unused hole in physical address space for alignment */
```

Comment:
You mean entire phys_rsv is now an unused
hole for alignment?

Reply:
what I mean here is that the 'U' mapping's physical address is aligned at the next available location above memsize (`=config.dpmi_base + dpmi_mem_size()`). So there's a hole between memsize and the physical location (guest_phys_addr in KVM) that is mapped to `&mem_base[0x110000]`.